flink-kubernetes/src/main/java/org/apache/flink/kubernetes/highavailability/KubernetesLeaderElectionHaServices.java

```diff
@@ -202,11 +199,7 @@ public void internalCleanup() throws Exception {
             exception = e;
         }
 
-        kubeClient
-                .deleteConfigMapsByLabels(
-                        KubernetesUtils.getConfigMapLabels(
-                                clusterId, LABEL_CONFIGMAP_TYPE_HIGH_AVAILABILITY))
-                .get();
```

Comment:
Based on your comment, the behaviour of this call changed now. 
Previously it would delete both configmaps (cluster + jobId) as they have the same label. But now only the cluster info ConfigMap will be deleted. Is that correct?

Reply:
For the code itself, yes, this PR changed the logic. However, for the internal of Flink, we do not change the behavior, and I think current change is the correct thing.

After each job finished, Flink will clean up job related data (such as checkpoint id counter) on [Dispatcher#removeJob](https://github.com/apache/flink/blob/278504a2787a154faf6f6401028d4bbadafbba0a/flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java#L1258), and once the cluster shutdown, it will then clean the cluster related configmap (such as HA related leader) on [ClusterEntrypoint#stopClusterServices](https://github.com/apache/flink/blob/278504a2787a154faf6f6401028d4bbadafbba0a/flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypoint.java#L502).

For `KubernetesLeaderElectionHaServices`, it should only clean cluster related configs on `#internalCleanup` instead of cleanning the configmaps via labels.